hooks/tk-multi-publish2/basic/publish_renders.py

```diff
@@ -0,0 +1,510 @@
+        if not work_paths:
+            self.logger.debug(
+                "'{}': No generated work files".format(node_name)
+            )
+            return {"accepted": False, "visible": True, "enabled": True, "checked": False}
+
+        publish_paths = self.sgtk.abstract_paths_from_template(publish_template, fields)
+
+        work_paths_to_publish = sorted(work_paths, reverse=True)
+        publish_fields = {}
+        for work_path in work_paths:
+            work_fields = work_template.get_fields(work_path)
+            self.logger.debug("Checking '{}'".format(work_path))
+            for publish_path in publish_paths:
+                # cache result as this might be slow
+                if publish_path not in publish_fields:
+                    publish_fields[publish_path] = publish_template.get_fields(publish_path)
+                publish_fields = publish_fields[publish_path]
+                if os.path.exists(publish_path) and publish_fields["version"] == work_fields["version"]:
+                    self.logger.debug("'{}' already published".format(work_path))
+                    # publish version exists
+                    work_paths_to_publish.remove(work_path)
+                    break
+
+        if not work_paths_to_publish:
+            self.logger.debug(
+                "'{}': All versions published".format(node_name)
+            )
+            return {"accepted": False, "visible": True, "enabled": False, "checked": False}
+
+        render_paths = self._get_all_render_node_paths()
+        if path not in self.__render_paths:
+            self.logger.debug(
+                "'{}' isn't connected to a render node".format(node_name)
+            )
+            return {"accepted": True, "visible": True, "enabled": False, "checked": False}
+
+        settings_dict = {
+            "work_paths": work_paths_to_publish,
+            "to_publish": work_paths_to_publish[0],
+        }
+        self._set_item_settings(settings_dict, settings)
+
+        return {
+            "accepted": True,
+            "checked": self.__render_paths[path]
+        }
+    
+    @staticmethod
+    def _get_template(template_name):
+        """
+        Get the template object from the given name.
+
+        :param template_name: The name of the template to retrieve.
+        :returns: The template object.
+        """
+        engine = sgtk.platform.current_engine()
+        return engine.get_template_by_name(template_name)
+
+    def validate(self, settings, item):
+        """
+        Validates the given item to check that it is ok to publish. Returns a
+        boolean to indicate validity.
+
+        :param settings: Dictionary of Settings. The keys are strings, matching
+            the keys returned in the settings property. The values are `Setting`
+            instances.
+        :param item: Item to process
+        :returns: True if item is valid, False otherwise.
+        """
+        node_name = item.properties["node"].getName()
```

Comment:
Unused `node_name` here... wait, maybe this should at top of `def accepted()` instead?